components/bt/bluedroid/btc/profile/std/a2dp/btc_media_task.c

```diff
@@ -179,11 +179,11 @@ void btc_a2dp_sink_reg_data_cb(esp_a2d_data_cb_t callback)
     bt_av_sink_data_callback = callback;
 }
 
-static inline void btc_a2d_data_cb_to_app(const uint8_t *data, uint32_t len)
+static inline void btc_a2d_data_cb_to_app(const uint8_t *data, uint32_t len, uint32_t sample_rate)
 {
     // todo: critical section protection
     if (bt_av_sink_data_callback) {
-        bt_av_sink_data_callback(data, len);
```

Comment:
For SBC codec, the sample frequency information can be provided in Codec Specific Information Element through ESP_A2D_AUDIO_CFG_EVT, so there is no need to transfer this parameter in the data callback function.

Reply:
I understand, however A2DP official spec says the following:
>The codec information that determines the bit rate is contained in the SBC frame
header and repeatedly sent to the SNK associated with audio data stream. The SRC
is capable of changing the bit rate dynamically by changing the bitpool parameter
without suspending.

Both 44.1 kHz and 48 kHz sample rate support is mandatory for audio sink
and sample rate can be changed dynamically without pausing the stream
so I saw no other way than to include it in the data callback.

Reply:
I think "bit rate" and "sampling frequency" have different meanings in the A2DP spec. "Sampling Frequency" is the raw sampled data stream which can be configured to be 44.1KHz, 48KHz, etc. "bit rate" indicates the encoded media stream(SBC frames) format which is transmitted over the air, this as you say, can be altered by A2DP source and is related to "bitpool" value as well as "Sampling Frequency."
Sampling frequency and Max/Min SBC bitpool value are determined in the AVDTP signaling procedure, they will not be changed unless reconfigured. Sampling frequency is already provided through "ESP_A2D_AUDIO_CFG_EVT", Since the output of ESP A2DP data callback is the decoded audio stream (PCM), and we don't need to output the bitrate here.

Reply:
>Sampling frequency is already provided through "ESP_A2D_AUDIO_CFG_EVT".

It is? I did not realise that, my mistake.